### ISSUE TITLE
Update docs to indicate that 'site' module is not supported by pyinstaller

### ISSUE THREAD

**Issue by luc-x41** [pyinstaller/pyinstaller#6419]
`exit()` in python code works when you run it with any python version that I know of, and pyinstaller builds it fine, but then crashes on runtime.

`sys.exit()` and `print()` work fine so it's not stdlib that is the problem in general.

Glancing at the docs, it says something about a `site` module being automatically loaded that provides it. Might it be that pyinstaller should load that also for full compatibility with normal python runtimes?

### Context information (for bug reports)

* Output of `pyinstaller --version`: 4.7
* Version of Python: 3.8.10
* Platform: GNU/Linux (Ubuntu 20.04.2 LTS)
* How you installed Python: apt install
* Did you also try this on another platform? No.
* try the latest development version. Same problem.
* follow *all* the instructions in our "If Things Go Wrong" Guide. Glancing over it, I don't see a solution there.

### Make sure [everything is packaged correctly](https://github.com/pyinstaller/pyinstaller/wiki/How-to-Report-Bugs#make-sure-everything-is-packaged-correctly)

  * [x] start with clean installation
  * [x] use the latest development version
  * [x] Run your frozen program **from a command window (shell)** — instead of double-clicking on it
  * [x] Package your program in **--onedir mode**
  * [x] Package **without UPX**, say: use the option `--noupx` or set `upx=False` in your .spec-file
  * [x] Repackage you application in **verbose/debug mode**. For this, pass the option `--debug=all` to `pyinstaller`

### A minimal example program which shows the error

```
exit()
```

### Stacktrace / full error message

```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
    exit()
NameError: name 'exit' is not defined
[2095841] Failed to execute script 'test' due to unhandled exception!
```

**Comment by rokm** [comment]
PyInstaller does not support `site` module and site configuration, so its operation mode is equivalent to that of `python -S`. Hence `site`-provided built-ins are not available. 

And you shouldn't be using them in your program, anyway - see the [docs](https://docs.python.org/3/library/constants.html#constants-added-by-the-site-module): "The site module (which is imported automatically during startup, except if the `-S` command-line option is given) adds several constants to the built-in namespace. **They are useful for the interactive interpreter shell and should not be used in programs.**"

**Comment by luc-x41** [comment]
> PyInstaller does not support `site` module and site configuration

Fair enough. Is this documented somewhere, so one knows that pyinstaller is not identical to running the code with regular cpython?

**Comment by rokm** [comment]
> > PyInstaller does not support `site` module and site configuration
> 
> Fair enough. Is this documented somewhere, so one knows that pyinstaller is not identical to running the code with regular cpython?

Probably not, and perhaps it should be.

Although on the other hand, the program should not be relying on presence of `site`-installed built-ins in the first place, because nothing prevents user from running it under `python -S` (which is also regular cpython)...

**Comment by luc-x41** [comment]
Makes sense. I was honestly unaware of this `site` module being auto-imported when starting to write the ticket, I must have learned from the python REPL and never saw that the python docs say one should not use `exit()`. Learned something today :)

I updated the ticket title to better reflect the new status.

**Comment by bwoodsend** [comment]
Where in the documentation would you expect to find this piece of information? It's a bit miscellaneous.

**Comment by rokm** [comment]
> Where in the documentation would you expect to find this piece of information? It's a bit miscellaneous.

I suppose we could add a section called "Runtime behavior and caveats" and document this. As well as the fact that module-swapping trick doesn't work for pure-python modules that are collected into PYZ.

**Comment by luc-x41** [comment]
I'm not too familiar with how `pyinstaller` works but I assume you must be parsing the code already for imports and whatnot. Maybe it makes sense to show a warning when a `quit` or `exit` call is found? Then people would see it when it's relevant to them.

Or maybe it's just too obscure a case (I was surprised there was no ticket about this yet, as setting exit statuses is not uncommon -- until I learned that you're not supposed to use it without importing `sys`) and people can find it in the issues now.

**Comment by bwoodsend** [comment]
> Maybe it makes sense to show a warning when a quit or exit call is found? Then people would see it when it's relevant to them.

That sounds like a very slippery slope to me. There are so many similar non-portable quirks where people do something wrong (e.g. assuming the current working directory is where just your code is, accessing raw .py files, storing configuration files inside their package instead of somewhere they're guaranteed write access, assuming docstrings will exist at runtime, expecting a full shell environment in a Windowed macOS app, trying to multiprocess by using `subprocess.run(["python", "sub_process.py"])`, assuming Python is available in `$PATH`, [assuming that entrypoints are available in `$PATH`](https://github.com/pyinstaller/pyinstaller/issues/6362), ...) that I reckon we could add checks until forever without really scratching the surface.
